Resolver/EntityAnnotationResolver.php

```diff
@@ -218,6 +218,11 @@ public function evaluateMapping(AnnotationEntity $annotation, $entity)
 
             if (!($instance instanceof $entityClass)) {
 
+                $notFoundException = $annotation->getNotFoundException();
+                if (!empty($notFoundException)) {
+                    throw new $notFoundException['exception']($notFoundException['message']);
```

Comment:
I'm not sure what are you talking about, maybe you've not read the code correctly or maybe I don't see what you're pointing :)

I'm taking an array, accessing one of it's value that contains an exception name to do a "new" of this exception name.

I've tested one code similar to that and seems to be working on 5.3.3: http://3v4l.org/hvCiV#v510